cms/djangoapps/import_from_modulestore/models.py

```
@@ -0,0 +1,172 @@
+
+    source_key = LearningContextKeyField(help_text=_('The modulestore course'), max_length=255, db_index=True)
+    target = models.ForeignKey(LearningPackage, models.SET_NULL, null=True)
+
+    def __str__(self):
+        return f'{self.source_key} - {self.target}'
+
+    def ready(self) -> None:
+        """
+        Set import status to ready.
+        """
+        self.status = ImportStatus.READY
+        self.save()
+
+    def imported(self) -> None:
+        """
+        Set import status to imported and clean related staged content.
+        """
+        self.status = ImportStatus.IMPORTED
+        self.save()
+        self.clean_related_staged_content()
+
+    def cancel(self) -> None:
+        """
+        Cancel import action and delete related staged content.
+        """
+        self.status = ImportStatus.CANCELED
+        self.save()
+        self.clean_related_staged_content()
+
+    def error(self) -> None:
+        """
+        Set import status to error and clean related staged
+        """
+        self.status = ImportStatus.ERROR
+        self.save()
+        self.clean_related_staged_content()
+
+    def clean_related_staged_content(self) -> None:
+        """
+        Clean related staged content.
+        """
+        for staged_content_for_import in self.staged_content_for_import.all():
+            staged_content_for_import.staged_content.delete()
+
+    class Meta:
+        verbose_name = _('Import from modulestore')
+        verbose_name_plural = _('Imports from modulestore')
+
+    @classmethod
+    def get_by_uuid(cls, import_uuid: str) -> Self | None:
+        """
+        Get an import task by its ID.
+        """
+        return cls.objects.filter(uuid=import_uuid).first()
+
+    @classmethod
+    def get_ready_by_uuid(cls, import_uuid: str) -> Self | None:
+        """
+        Get an import task by its UUID.
+        """
+        return cls.objects.filter(uuid=import_uuid, status=ImportStatus.READY).first()
+
```

These methods are not necessary. They are just wrapping simple operations which Django ORM already provides us. 

They are also quietly suppressing an error case: the case when there is no matching instance. This is un-pythonic. When code needs to query for a particular model instance, it is most idiomatic for the code to explicitly call `.objects.get(...)`, and then explicitly handle `Import.DoesNotExist`.

```suggestion
```

---

I agree with you, I have removed these methods.